inst/@symfun/minus.m

```diff
@@ -74,14 +74,14 @@
 %! f(x) = x^2;
 %! g = x^2;
 %! s = warning('off', 'OctSymPy:sym:arithmetic:workaround42735');
-%! h = x - f;  assert(isa(h, 'symfun') && isequal(h.sym, x - g))
-%! h = x + f;  assert(isa(h, 'symfun') && isequal(h.sym, x + g))
-%! h = x * f;  assert(isa(h, 'symfun') && isequal(h.sym, x * g))
-%! h = x / f;  assert(isa(h, 'symfun') && isequal(h.sym, x / g))
-%! h = x ^ f;  assert(isa(h, 'symfun') && isequal(h.sym, x ^ g))
-%! h = x .* f; assert(isa(h, 'symfun') && isequal(h.sym, x .* g))
-%! h = x ./ f; assert(isa(h, 'symfun') && isequal(h.sym, x ./ g))
-%! h = x .^ f; assert(isa(h, 'symfun') && isequal(h.sym, x .^ g))
+%! h = f - x;  assert(isa(h, 'symfun') && isequal(h.symbol, g - x))
+%! h = f + x;  assert(isa(h, 'symfun') && isequal(h.symbol, g + x))
+%! h = f * x;  assert(isa(h, 'symfun') && isequal(h.symbol, g * x))
+%! h = f / x;  assert(isa(h, 'symfun') && isequal(h.symbol, g / x))
+%! h = f ^ x;  assert(isa(h, 'symfun') && isequal(h.symbol, g ^ x))
+%! h = f .* x; assert(isa(h, 'symfun') && isequal(h.symbol, g .* x))
+%! h = f ./ x; assert(isa(h, 'symfun') && isequal(h.symbol, g ./ x))
+%! h = f .^ x; assert(isa(h, 'symfun') && isequal(h.symbol, g .^ x))
```

Comment:
These tests should stay `sym - symfun`: the whole point of this test is the Octave bug #42735.  You can mark it `xtest` if you can't make it work rather than changing it.

I notice the older workaround gives infinite recursion so we may need a new workaround!  In the workaround in `@sym/plus`, I tried `plus@symfun(x, y)` but that gave an error message than `symfun` is not a subclass of `sym` (which is true enough).

In some cases, such as ``@sym/minus.m` we could do:
````
  % Dear hacker from the distant future... maybe you can delete this?
  if (isa(y, 'symfun'))
    warning('OctSymPy:sym:arithmetic:workaround42735', ...
            'worked around octave bug #42735')
    z = plus(-y, x);
    return
  end
````
(but I don't know we can do that for every case).